src/prefect/engine/cloud/task_runner.py

```diff
@@ -46,16 +46,9 @@ class CloudTaskRunner(TaskRunner):
             retrieving and storing state results during execution (if the Task doesn't already have one)
     """
 
-    def __init__(
-        self,
-        task: Task,
-        state_handlers: Iterable[Callable] = None,
-        result_handler: ResultHandler = None,
-    ) -> None:
+    def __init__(self, task: Task, state_handlers: Iterable[Callable] = None) -> None:
         self.client = Client()
-        super().__init__(
```

Comment:
We ended up deciding that these actually need to stay because this is how global configuration of result handlers is set by people. So I think I still need to add this back in (and also added back in the places below where I had taken passing it in out from any instantiation of a task runner)

Reply:
Yea this is the place where the _Flow_ can pass it's `Result` type down in case the `task` doesn't have one set.  So I think we should:
- expose `result` here and pass it down from the flow
- `self.result = self.task.result or result` at init
- replace `self.task.result` everywhere in the pipeline with `self.result`

Reply:
Reintroduced result_handler forwarding (for backwards compatability/conversion) in af59d67 and added in result forwarding/`self.result` instead of `self.task.result` stuff in e1f8e2f